### PR TITLE
Add support for themes

### PR DESCRIPTION
**Pull request by DirtyF**
Following the same principles as plugins : 

- [x]  _themes collection 
- [x] Github Pages Themes are listed in `_data/github_pages.yml`
- [x] Add GitHub Pages Themes
- [x] Add HTML5UP Themes by @andrewbanchich
- [x] Themes screenshots
- [x] 12 themes listed on the homepage with thumbnail

jekyll-directory-index-desktop

- [x] Complete themes listing with screenshots

jekyll-directory-themes-desktop

- [x] Theme detail with large screenshot

jekyll-directory-theme-install-desktop

### CODE REVIEW

**Review by pathawks — APPROVED** [review]
:heart:

File: _config.yml [hunk 1]
```diff
@@ -1,16 +1,18 @@
-title: "Jekyll Plugins • Community-built plugins to level up your Jekyll experience."
+title: "Jekyll directory • Community-built themes and plugins to level up your Jekyll experience."
 repository: https://github.com/jekyll/plugins
 url: http://jekyll.github.io
-baseurl: /plugins
+baseurl: /directory
```
Comment (pathawks) [inline _config.yml]: ~~This seems to have broken everything.~~

Do we need to rename the repository at the same time we make this change?

**Review by pathawks — CHANGES_REQUESTED** [review]

**Comment by DirtyF** [comment]
@jekyll/plugin-core added only Github Pages themes for now, will add some gem-based themes later. see https://github.com/jekyll/plugins/pull/33#issue-201442758 for a rendering preview.

File: _config.yml (hunk 1 above)
Comment (parkr) [reply _config.yml]: I would let GitHub Pages do this for you automatically: https://github.com/jekyll/github-metadata/blob/8906f2b9c890f0aafef96423c7cd7e5047f7dae4/lib/jekyll-github-metadata/ghp_metadata_generator.rb#L36-L45

File: _config.yml [hunk 2]
```diff
@@ -20,12 +22,19 @@ defaults:
       layout: plugin              # DO NOT CHANGE THIS DEFAULT
       type: rubygems              # DO NOT CHANGE THIS DEFAULT
 
+  - scope:
+      type: themes
+    values:
+      permalink: /themes/:title/  # DO NOT CHANGE THIS DEFAULT
+      layout: theme               # DO NOT CHANGE THIS DEFAULT
+      type: rubygems              # DO NOT CHANGE THIS DEFAULT
```
Comment (parkr) [inline _config.yml:31]: Idea: let's have a test spec that explicitly checks that the individual themes (and plugins above) abide by these conventions unless explicitly overridden in the case of `type` only.

**Comment by DirtyF** [comment]
Preview on Github Pages on my repo: https://dirtyf.github.io/directory/

**Review by parkr — APPROVED** [review]
This diff is way too huge to review so merge once you think it's ready! We can always iterate 😄

**Pull request merged**